app/controllers/publications_controller.rb

```diff
@@ -0,0 +1,76 @@
+class PublicationsController < ApplicationController
+
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :set_publication, only: [:show, :edit, :update, :destroy]
+
+  # GET /publications
+  # GET /publications.json
+  def index
+    @publications = Publication.all
+  end
+
+  # GET /publications/1
+  # GET /publications/1.json
+  def show
+  end
+
+  # GET /publications/new
+  def new
+    @publication = Publication.new
+  end
+
+  # GET /publications/1/edit
+
+  # POST /publications
+  # POST /publications.json
+  def create
+    @publication = Publication.new(publication_params)
+    authorize @publication
+    @user_publication = UserPublication.new
```

Comment:
can do this after publication is saved. so inside the if saved condition. why init it here when we might not be using it